src/components/frame/v5/pages/AgreementsPage/partials/AgreementCard/AgreementCard.tsx

```diff
@@ -0,0 +1,236 @@
+import { MotionState as NetworkMotionState } from '@colony/colony-js';
+import clsx from 'clsx';
+import { format } from 'date-fns';
+import { BigNumber } from 'ethers';
+import { FilePlus, ShareNetwork } from 'phosphor-react';
+import React, { useState, type FC, useEffect, useMemo } from 'react';
+import { generatePath, useNavigate } from 'react-router-dom';
+
+import MeatballMenuCopyItem from '~common/ColonyActionsTable/partials/MeatballMenuCopyItem/MeatballMenuCopyItem.tsx';
+import { useColonyContext } from '~context/ColonyContext.tsx';
+import useUserByNameOrAddress from '~hooks/useUserByNameOrAddress.ts';
+import {
+  COLONY_AGREEMENTS_ROUTE,
+  COLONY_HOME_ROUTE,
+  TX_SEARCH_PARAM,
+} from '~routes';
+import { getMotionState, MotionState } from '~utils/colonyMotions.ts';
+import { getEnumValueFromKey } from '~utils/getEnumValueFromKey.ts';
+import { formatText } from '~utils/intl.ts';
+import { getSafePollingInterval } from '~utils/queries.ts';
+import { useGetColonyAction } from '~v5/common/ActionSidebar/hooks/useGetColonyAction.ts';
+import MotionCountDownTimer from '~v5/common/ActionSidebar/partials/Motions/partials/MotionCountDownTimer/MotionCountDownTimer.tsx';
+import {
+  type Steps,
+  CustomStep,
+} from '~v5/common/ActionSidebar/partials/Motions/types.ts';
+import MotionStateBadge from '~v5/common/Pills/MotionStateBadge/MotionStateBadge.tsx';
+import TeamBadge from '~v5/common/Pills/TeamBadge/TeamBadge.tsx';
+import Avatar from '~v5/shared/Avatar/Avatar.tsx';
+import MeatBallMenu from '~v5/shared/MeatBallMenu/MeatBallMenu.tsx';
+import UserPopover from '~v5/shared/UserPopover/UserPopover.tsx';
+
+import { type AgreementCardProps } from './types.ts';
+import { renderAgreementCardSkeleton } from './utils.tsx';
+
+const AgreementCard: FC<AgreementCardProps> = ({ transactionId }) => {
+  const {
+    action,
+    refetchMotionState,
+    loadingAction,
+    motionState,
+    startPollingForAction,
+    stopPollingForAction,
+  } = useGetColonyAction(transactionId);
```

Comment:
Could we check how many actual backend calls we're doing with this? 
Maybe we could add a `cache-first` fetchPolicy to the `useGetColonyAction` hook, since we're fetching all the actions and then refetching each one so we can poll if they are active.